### PR TITLE
feat: Mint and Burn tokens

### PR DESCRIPTION
**Pull request by stanleyjones**
Related:
- liftedinit/roadmap#37
- liftedinit/roadmap#38
- liftedinit/gwen#43

### CODE REVIEW

File: src/network/modules/ledger/ledger.ts [hunk 1]
```diff
@@ -10,8 +10,6 @@ interface Ledger extends NetworkModule {
   _namespace_: string
   info: () => Promise<LedgerInfo>
   balance: (address?: string, symbols?: string[]) => Promise<Balances>
-  mint: () => Promise<unknown>
-  burn: () => Promise<unknown>
```
Comment (stanleyjones) [inline src/network/modules/ledger/ledger.ts:14]: `mint` and `burn` are no longer in the `ledger` namespace (for a long time now).

File: src/network/modules/tokens/__tests__/tokens.test.ts [hunk 2]
```diff
@@ -69,14 +69,38 @@ describe("Tokens", () => {
...
+    it("should burn some tokens", async () => {
+      const mockCall = jest.fn(async () => { })
+      const tokens = setupModule(Tokens, mockCall)
+
+      const symbol = mockTokenString
+      const addresses = { maa: BigInt(1000) }
+      tokens.burn({ symbol, addresses })
+
       expect(mockCall).toHaveBeenCalled()
     })
```
Comment (stanleyjones) [inline src/network/modules/tokens/__tests__/tokens.test.ts:105]: Yes, these tests suck.

File: src/network/modules/tokens/tokens.ts [hunk 3]
```diff
@@ -117,6 +133,13 @@ function makeTokensRemoveExtendedData(
   return data
 }
 
+function makeTokensMintBurnParam(param: TokensMintBurnParam) {
+  const data = new Map()
+  data.set(0, param.symbol)
+  data.set(1, new Map(Object.entries(param.addresses)))
+  return data
+}
```
Comment (stanleyjones) [inline src/network/modules/tokens/tokens.ts:141]: Thank you @hansl for making the `@param` for these two methods the same. Much appreciated!

File: src/network/modules/tokens/types.ts [hunk 4]
```diff
@@ -61,6 +61,15 @@ export interface TokensRemoveExtendedParam {
   indices: AttrIndex[]
 }
 
+export interface TokensAddressMap {
+  [address: string]: bigint
+}
+
+export interface TokensMintBurnParam {
+  symbol: string
+  addresses: TokensAddressMap
```
Comment (stanleyjones) [inline src/network/modules/tokens/types.ts:70]: ManyJS will let you send multiple addresses and amounts even if Gwen doesn't yet.

**Review by fmorency — CHANGES_REQUESTED** [review]
Minor thing. Thanks @stanleyjones!

File: src/network/modules/tokens/tokens.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (stanleyjones) [reply src/network/modules/tokens/tokens.ts]: Thanks!

**Review by hansl — APPROVED** [review]

**Pull request merged**